mlir/lib/CAPI/Dialect/Rock.cpp

```diff
@@ -0,0 +1,342 @@
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  newSpace->numHeuristicQuick = 0;
+}
+
+void createConv2DFwdTuningRangeMLR(struct rock::TunableParams *newSpace,
+                                   rock::Conv2DOp convOp) {
+  // Initial point determined by Multi Linear Regression coefficients and
+  // iterate further on.
+
+  // These needs to have separate analysis for their own coefficient.
+  // For now, NCHW/F32/FWD only.
+  uint64_t layout = 0; // NCHW
+  uint64_t data_type;
+  uint64_t direction;
+
+  uint64_t in_channels;
+  uint64_t in_h;
+  uint64_t in_w;
+  uint64_t fil_h;
+  uint64_t fil_w;
+  uint64_t out_channels;
+  uint64_t batchsize;
+  uint64_t group_count;
+
+  auto input = convOp->getOperand(0);
+  auto weight = convOp->getOperand(1);
+  ShapedType inTy = input.getType().cast<ShapedType>();
+  auto inShape = inTy.getShape();
+  batchsize = inShape[0];
+
+  ShapedType wtTy = weight.getType().cast<ShapedType>();
+  auto wtShape = wtTy.getShape();
+  out_channels = wtShape[0];
+
+  if (layout == 0) { // NCHW
+    in_channels = inShape[1];
+    in_h = inShape[2];
+    in_w = inShape[3];
+    fil_h = wtShape[2];
+    fil_w = wtShape[3];
+  } else { // NHWC
+    in_channels = inShape[3];
+    in_h = inShape[2];
+    in_w = inShape[3];
+    fil_h = wtShape[2];
+    fil_w = wtShape[3];
+  }
+  group_count = inShape[4];
+
+  rock::ConvolutionContext ctx = rock::populateConvContext(convOp);
+  int64_t leftPadH = ctx.getPaddingVal()[0];
+  int64_t leftPadW = ctx.getPaddingVal()[2];
+  int64_t rightPadH = ctx.getPaddingVal()[1];
+  int64_t rightPadW = ctx.getPaddingVal()[3];
+
+  int64_t dilationH = ctx.getDilationVal()[0];
+  int64_t dilationW = ctx.getDilationVal()[1];
+  int64_t strideH = ctx.getStrideVal()[0];
+  int64_t strideW = ctx.getStrideVal()[1];
+
+  std::vector<std::vector<uint32_t>> tParams = {{4, 8, 16, 32, 64, 128},
+                                                {16, 32, 64, 128},
+                                                {16, 32, 64, 128},
+                                                {16, 32, 64},
+                                                {16, 32, 64},
+                                                {1, 4},
+                                                {0, 1},
+                                                {0, 1}};
+  // std::vector<uint32_t> p7b = {1, 2, 4, 8};
+
+  // Each field of tuning parameter needs one set of coefficient + 1 constant.
+  // ((16+1)*8) number of quick set can be also set by MLR
+
+  // clang-format off
+  //                                         in_ch ;   in_h;   in_w;  fil_h;  fil_w; out_ch;  batch;  lPadH;  lPadW;  rPadH;  rPadW;   strH;   strW;   dilH;   dilW;  group;  const;
+  // From gfx90878_1.1.0.udb tuned at 9.Oct.2022  Weekly CI xdlops
+  std::vector<std::vector<float>> coeffs = {{ 0.0175,-1.1229,-1.1229,-12.153,-12.153,-0.0413, 0.0000,47.5203,47.5203,47.5203,47.5203,55.0025,55.0025, 0.0000, 0.0000, 0.0000, 108.5339},
```

Comment:
This'll be architecture-dependent, make sure `arch` is a parameter somehow.

And probably include a script to generate these.